bigframes/session/__init__.py

```diff
@@ -177,6 +177,7 @@ def __init__(
         # Now that we're starting the session, don't allow the options to be
         # changed.
         context._session_started = True
+        self._df_snapshot: Dict[bigquery.TableReference, datetime.datetime] = {}
```

Comment:
Just curious, is TableReference hashable to be used as the dict key?

Reply:
Then could you help check if TableReference will change when there is new data coming in or anything happened to the layout (e.g., partitioning).

Can we use the full table ID/path as the key to be insensitive to such changes?